### ISSUE TITLE
Cannot use reserved words as edges when you specify a storage key

### ISSUE THREAD

**Issue by imle** [facebook/ent#918]
As an example:
```
edge.To("game_map", GameStats.Type).StorageKey(edge.Column("map")),
```

The call in the template that seems to end up creating the issue is `StructFKField`.

I am not very familiar with the codebase or I would put in a PR, but I can if no one has the time to look at this.

**Comment by a8m** [comment]
Hey @imle and thanks for reporting this issue.

> I am not very familiar with the codebase or I would put in a PR, but I can if no one has the time to look at this.

If you want to contribute this fix, the change should be simple. The column name needs to be wrapped with the `builderField` function to avoid conflicts with Go keywords (and other internal fields):

1\. Change the field name in FK creation (see [code pointer](https://github.com/facebook/ent/blob/fb97ca66ac4b1cfe0b85f4f060c3996d55ca7c22/entc/gen/type.go#L548)):
```diff
- Name:        e.Rel.Column(),
+ Name:        builderField(e.Rel.Column()),
```

2. Use the same name in `StructFKField` (see [code pointer](https://github.com/facebook/ent/blob/fb97ca66ac4b1cfe0b85f4f060c3996d55ca7c22/entc/gen/type.go#L1133)):
```diff
- return e.Rel.Column()
+ return builderField(e.Rel.Column())
```

**Comment by a8m** [comment]
Closed with #922
